api-samples/declarativeNetRequest/no-cookies-manager/manager.js

```diff
@@ -0,0 +1,190 @@
+  ruleItem.querySelector('.case-sensitive').checked = caseSensitive;
+
+  // Set the conditionValue input
+  ruleItem.querySelector('.condition-value').value = conditionValue;
+
+  // Set button click handlers
+  function getCurrentRuleValues() {
+    const _conditionValue = ruleItem.querySelector('.condition-value').value;
+    const _conditionType = ruleItem.querySelector('.condition-type').value;
+    const _caseSensitive = ruleItem.querySelector('.case-sensitive').checked;
+    return {
+      conditionValue: _conditionValue.trim(),
+      conditionType: _conditionType,
+      caseSensitive: _caseSensitive
+    };
+  }
+  ruleItem.querySelector('.remove-rule').addEventListener('click', async () => {
+    // For a new rule, just remove the item from the DOM
+    if (id !== 'NEW') {
+      await removeRule(id);
+    }
+    ruleItem.remove();
+  });
+  ruleItem.querySelector('.save-rule').addEventListener('click', async () => {
+    const {
+      conditionValue: _conditionValue,
+      conditionType: _conditionType,
+      caseSensitive: _caseSensitive
+    } = getCurrentRuleValues();
+    const _id = id === 'NEW' ? await getNextRuleId() : id;
+    await saveRule({
+      id: _id,
+      conditionType: _conditionType,
+      conditionValue: _conditionValue,
+      caseSensitive: _caseSensitive
+    });
+    // Update the rule ID
+    ruleItem.querySelector('.rule-id').value = _id;
+    // Disable the save button
+    setSaveButtonDisabled();
+    // Update parameter
+    id = _id;
+    conditionType = _conditionType;
+    conditionValue = _conditionValue;
+    caseSensitive = _caseSensitive;
+  });
+
+  // Disable the save button
+  function setSaveButtonEnabled() {
+    ruleItem.querySelector('.save-rule').disabled = false;
+  }
+  function setSaveButtonDisabled() {
+    ruleItem.querySelector('.save-rule').disabled = true;
+  }
+  setSaveButtonDisabled();
+
+  // Set input change handlers
+  async function verify() {
+    const {
+      conditionValue: _conditionValue,
+      conditionType: _conditionType,
+      caseSensitive: _caseSensitive
+    } = getCurrentRuleValues();
+
+    if (
+      conditionValue === _conditionValue &&
+      caseSensitive === _caseSensitive &&
+      conditionType === _conditionType
+    ) {
+      // If the rule is not changed, disable the save button
+      setSaveButtonDisabled();
+      return;
+    }
+    if (_conditionValue.trim() === '') {
+      // If the condition value is empty, disable the save button
+      setSaveButtonDisabled();
+      return;
+    }
+    // For the regex filter, verify if the regex is supported
+    if (_conditionType === 'regexFilter') {
+      const result = await chrome.declarativeNetRequest.isRegexSupported({
+        isCaseSensitive: _caseSensitive,
+        regex: _conditionValue
+      });
+      if (!result.isSupported) {
+        // If the regex is invalid, disable the save button
+        setSaveButtonDisabled();
+        alert(`Invalid regex: ${result.reason}`);
```

Comment:
Isn't this going to alert for every character you type? That seems like it might be a bit aggressive.

Reply:
I'm not sure, but I think the `change` event will only be triggered when the input box loses focus. The event that is triggered every time there is an input should be the `input` event.

Reply:
You're right! I thought I deleted this comment actually, but I guess I didn't.